examples/pytorch/ogb/ogbn-mag/hetero_rgcn.py

```diff
@@ -392,18 +397,36 @@ def main(args):
         )
         optimizer = th.optim.Adam(all_params, lr=0.01)
 
-        logger = train(
-            g,
-            model,
-            embed_layer,
-            optimizer,
-            train_loader,
-            split_idx,
-            labels,
-            logger,
-            device,
-            run,
-        )
+        if device == "cpu":
+            cores = list(range(os.cpu_count()))
+            len_cores = len(cores)
+            dl_cores = cores[:int(len_cores / 2)]
+            with train_loader.enable_cpu_affinity(dl_cores, [i for i in cores if i not in dl_cores]):
```

Comment:
Are you familiar with this practice? Could you take a look? @BarclayII @chang-l @yaox12

Reply:
@mufeili , I added a pytorch version check here, since enable_cpu_affinity is enabled in pytorch 1.12. CPU affinity is basically bind partial cpu only for dataloader and others for computing. 
![image](https://user-images.githubusercontent.com/4355494/194888625-64a89f9b-686e-492e-8871-736ac7b1e5e8.png)

Reply:
I have some questions/suggestions listed here: 
> since enable_cpu_affinity is enabled in pytorch 1.12  

(1)  It does not make sense to me... Since `enable_cpu_affinity` is a dgl method, why it has to be restricted to pytorch 1.12+?  Can you pls provide more info here why it fails with pytorch <1.12?

(2) I don't think using `enable_cpu_affinity` as `with train_loader.enable_cpu_affinity(dl_cores, [i for i in cores if i not in dl_cores]):` is a good practice for general users (too much low-level exposure). Would it be enough to use default `with dataloader.enable_cpu_affinity():` as long as `num_workers` is specified properly?

Reply:
@chang-l , enable_cpu_affinity is actually a feature of pytorch, and it appears to me that pytorch version smaller than 1.12 won't be able to recognize this API, and exception will be raised.
![image](https://user-images.githubusercontent.com/4355494/194940502-c833289c-d83a-4e7d-ae02-9b2ccbf14ee7.png)

And I found the default loader_cores and compute_cores are None, so no argument provided in with dataloader.enable_cpu_affinity() and set num_workers > 0 would also trigger an exception.
![image](https://user-images.githubusercontent.com/4355494/194940820-d2f66ead-f892-45f1-9e1c-421ecfe0e0da.png)

![image](https://user-images.githubusercontent.com/4355494/194940754-b86a15c0-c84a-4513-91e6-4e2ab7ffe2a0.png)

Reply:
For (1), are you sure you were using the same version of dgl for your pytorch 1.10 run? 
For (2), I think the crash is due to your setting of `num_of_workers = os.cpu_count/2`. According to the [doc](https://docs.dgl.ai/en/0.9.x/tutorials/cpu/cpu_best_practises.html), you have to make sure your compute_core + loader_core <= num_core_per_socket (single NUMA node).  I guess you were running on a 2x48 core cpu? If so, please make sure `num_workers<48`; or simply remove the assignment `num_workers = int(os.cpu_count() / 2)`, and use cmd instead.

Reply:
@chang-l , for (1), not exactly the same, but both are above 0.9.0. For 1.10, dgl is 0.9.0, and for 1.12, dgl is 0.9.1
(2) Yes, That is my way to prove why I used `train_loader.enable_cpu_affinity(dl_cores, [i for i in cores if i not in dl_cores])` instead of `with dataloader.enable_cpu_affinity()`, I found tuning num_workers from 0 to half of cores helped to improve CPU side performance roughly by 20%

Reply:
I think the discrepancy between 0.9.0 and 0.9.1 might explain your first crash. Can you somehow check your installed source code path by doing `pip show dgl` and see if `enable_cpu_affinity` is defined in SourcePath/dgl/dataloading/dataloader.py? I guess your 1.10 pytorch + 0.9.0(dgl) missing the def of such function.

For (2), I see... But you are using 2x48 cores (48 for load, 48 for compute) from both sockets to gain 1.2X, right?
Could you please perform another study on the scaling? I mean, comparing your 2x48 results to the single socket default `with dataloader.enable_cpu_affinity()` with `--num_workers 24`, assuming 48 is your nprocs_per_socket.

BTW, you need to remove your L381, to respect the cmd setting of num_workers

Reply:
@chang-l , thanks for the comment, I fixed the codes according to your above suggestions. And you're, dgl version is root cause. And num_workers as 24 also bring performance benefit, so I removed my hard code of obtain system num_cores and simply use input num_workers configuration.
![image](https://user-images.githubusercontent.com/4355494/195186454-e4556022-8260-4213-acf1-c3fe3d7e9b10.png)
![image](https://user-images.githubusercontent.com/4355494/195186503-f6154485-abfe-4993-a729-ecf2a1b95898.png)